src/Microsoft.AspNetCore.Http.Abstractions/Routing/EndpointHttpContextExtensions.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Http.Features;
+
+namespace Microsoft.AspNetCore.Http
```

Comment:
Why do we need this to show up in all applications?

Reply:
What do you want instead?

Reply:
Getting the endpoint is relatively easy. Setting requires creating a class that implements `IEndpointFeature`. Making people implement an interface to set an endpoint feels like a pit of failure.

Both of these helper methods makes it feel much easier:

`context.GetEndpoint()` and `context.SetEndpoint(endpoint)`

vs

 `context.Features.Get<IEndpointFeature>()?.Endpoint` and `context.Features.Set<IEndpointFeature>(new EndpointFeature { Endpoint = endpoint })`

Reply:
Can we change the namespace then?

Reply:
Quite a few middleware will be using this soon. We have sometimes move extensions out of the expected namespace as a means of de-emphasizing them. I don't see any harm in keeping this one, nor have any suggestions for an alternate namespace.

Reply:
I disagree. 4 middleware will be using it.  A brand new one that we wrote (Authorization), static files (maybe), CORS, Response caching. Putting extension methods on HttpContext. 

We should not have this by default on HttpContext in every ASP.NET Core application. It just doesn't need to be that first class.

Reply:
There is a Microsoft.AspNetCore.Http.Extensions project it could move to. The extension methods in it still live in the `Microsoft.AspNetCore.Http` namespace.

Reply:
It's the namespace that's the problem, not the package.

Actually I expect this to get used in most of the middleware if only to access the metadata. I know Authentication needs it, and likely response compression, buffering, etc..

Reply:
Change the namespace and I'll be happy.

Reply:
I can't think of another namespace. All the endpoint types live in Microsoft.AspNetCore.Http.

Reply:
Those extensions are worthy of being that top level, this one isn’t. Add something else to the namespace like Endpoints

Reply:
It would be alone in Microsoft.AspNetCore.Http.Endpoints but it makes the most amount of sense.

Reply:
Sure, go for it.